gcloud/datastore/helpers.py

```diff
@@ -106,16 +155,67 @@ def entity_from_protobuf(pb):
                                  'be indexed or all excluded from indexes.')
 
             if not indexed_values.pop():
-                exclude_from_indexes.append(property_pb.name)
+                exclude_from_indexes.append(prop_name)
         else:
             if not property_pb.value.indexed:
-                exclude_from_indexes.append(property_pb.name)
+                exclude_from_indexes.append(prop_name)
 
     entity = Entity(key=key, exclude_from_indexes=exclude_from_indexes)
     entity.update(entity_props)
+    entity._meanings.update(entity_meanings)
     return entity
 
 
+def entity_to_protobuf(entity):
+    """Converts an entity into a protobuf.
+
+    :type entity: :class:`gcloud.datastore.entity.Entity`
+    :param entity: The entity to be turned into a protobuf.
+
+    :rype: :class:`gcloud.datastore._datastore_v1_pb2.Entity`
+    :returns: The Protobuf representing the entity.
+    """
+    entity_pb = datastore_pb.Entity()
+    if entity.key is not None:
+        key_pb = entity.key.to_protobuf()
+        entity_pb.key.CopyFrom(key_pb)
+
+    for name, value in entity.items():
+        value_is_list = isinstance(value, list)
+        if value_is_list and len(value) == 0:
+            continue
+
+        prop = entity_pb.property.add()
+        # Set the name of the property.
+        prop.name = name
+
+        # Set the appropriate value.
+        _set_protobuf_value(prop.value, value)
+
+        # Add index information to protobuf.
+        if name in entity.exclude_from_indexes:
+            if not value_is_list:
+                prop.value.indexed = False
+
+            for sub_value in prop.value.list_value:
+                sub_value.indexed = False
+
+        # Add meaning information to protobuf.
+        if name in entity._meanings:
+            meaning, orig_value = entity._meanings[name]
+            # Only add the meaning back to the protobuf if the value is
+            # unchanged from when it was originally read from the API.
+            if orig_value is value:
```

Comment:
@tseaver This is "broken" but I sent the PR because I figured two brains would be better than one.

The idea is

> Preserve the meaning unless the user changes the value

but this doesn't work if `orig_value` is a mutable data type (e.g. a `list` or an `Entity` instance).

I'd rather just override `__setattr__`, `__delattr__` and whatever else I need to to allow us to check edits.